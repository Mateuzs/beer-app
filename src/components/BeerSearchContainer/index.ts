export { default as BeerSearchContainer } from "./BeerSearchContainer";
