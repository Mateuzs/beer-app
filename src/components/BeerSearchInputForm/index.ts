export { default as BeerSearchInputForm } from "./BeerSearchInputForm";
