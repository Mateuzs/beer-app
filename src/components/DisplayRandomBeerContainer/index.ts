export { default as DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
