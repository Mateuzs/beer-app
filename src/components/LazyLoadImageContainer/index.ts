export { default as LazyLoadImageContainer } from "./LazyLoadImageContainer";
