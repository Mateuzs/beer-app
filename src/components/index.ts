export { BeerSearchContainer } from "./BeerSearchContainer";
export { DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
export { NavBar } from "./NavBar";
export { SearchButton } from "./SearchButton";
export { LazyLoadImageContainer } from "./LazyLoadImageContainer";
export { ErrorMessage } from "./ErrorMessage";
export { BeerSearchInputForm } from "./BeerSearchInputForm";
export { BeerList } from "./BeerList";
