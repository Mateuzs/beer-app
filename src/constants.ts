// in this simplified solution all wording has been defined in constants
// in real scenario it's required a translate mechanism to support many languages

export const APP_HEADER = "Beer App";

export const ERROR_MESSAGE = "Something went wrong! Try again later.";

export const LABEL_SEARCH_BEER = "Another beer!";
export const LABEL_SEARCH_NON_ALCOHOLIC_BEER = "Random non alcoholic beer";
export const LOADING_INFO = "Loading...";
export const IMAGE_NOT_AVAILABLE_INFO = "Image not available.";
export const IMAGE_URL_PLACEHOLDER = "Loading...";
export const LAZY_IMAGE_BLUR_EFFECT = "blur";
export const BEER_IMAGE_HEIGHT_PX = 300;
export const SMALL_BEER_IMAGE_HEIGHT_PX = 150;

export const BEER_SEARCH_INPUT_DATE_DESCRIPTION = "format date: MM-YYYY";
export const BEER_SEARCH_INPUT_NAME_DESCRIPTION = "Only letters and numbers";
export const BEER_SEARCH_INPUT_DATE_PLACEHOLDER = "Date";
export const BEER_SEARCH_INPUT_NAME_PLACEHOLDER = "Name";

export const BEER_SEARCH_INPUT_RADIO_NAME = "beer-settings";
export const BEER_SEARCH_INPUT_RADIO_TYPE = "radio";
export const BEER_SEARCH_INPUT_RADIO_NAME_DESCRIPTION = "Name";
export const BEER_SEARCH_INPUT_RADIO_DATE_DESCRIPTION = "Brewed Before";
export const BEER_SEARCH_INFO = "Nothing to display";

export const SEARCH_BUTTON_LABEL = "Search!";
