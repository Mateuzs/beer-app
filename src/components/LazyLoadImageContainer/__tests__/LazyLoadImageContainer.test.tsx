// external
import React from "react";
import { shallow, mount } from "enzyme";
// components
import { LazyLoadImageContainer } from "../../../components";
// constants, utils
import { IMAGE_NOT_AVAILABLE_INFO, IMAGE_URL_PLACEHOLDER, LOADING_INFO } from "../../../constants";

describe("LazyLoadImageContainer component", () => {
  test("should match the snapshot and styles", () => {
    const mockedData = {
      imageUrl: "image-url",
      height: 100,
    };
    const wrapper = shallow(<LazyLoadImageContainer {...mockedData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should render lazy load image component when url provided", () => {
    const mockedData = {
      imageUrl: "image-url",
      height: 100,
    };
    const component = mount(<LazyLoadImageContainer {...mockedData} />);

    expect(component.find('[data-test-id="lazy-load-image"]').length).toEqual(1);
    expect(component.find('[data-test-id="lazy-load-image-placeholder"]').length).toEqual(0);
  });

  test("should render lazy load image placeholder with proper text when url is placeholder", () => {
    const mockedData = {
      imageUrl: IMAGE_URL_PLACEHOLDER,
      height: 100,
    };
    const component = mount(<LazyLoadImageContainer {...mockedData} />);

    expect(component.find('[data-test-id="lazy-load-image"]').length).toEqual(0);
    expect(component.find('[data-test-id="lazy-load-image-placeholder"]').length).toEqual(1);
    expect(component.find('[data-test-id="lazy-load-image-placeholder"]').text()).toEqual(
      LOADING_INFO
    );
  });

  test("should render lazy load image placeholder with proper text when url is not available", () => {
    const mockedData = {
      imageUrl: null,
      height: 100,
    };
    const component = mount(<LazyLoadImageContainer {...mockedData} />);

    expect(component.find('[data-test-id="lazy-load-image"]').length).toEqual(0);
    expect(component.find('[data-test-id="lazy-load-image-placeholder"]').length).toEqual(1);
    expect(component.find('[data-test-id="lazy-load-image-placeholder"]').text()).toEqual(
      IMAGE_NOT_AVAILABLE_INFO
    );
  });
});
