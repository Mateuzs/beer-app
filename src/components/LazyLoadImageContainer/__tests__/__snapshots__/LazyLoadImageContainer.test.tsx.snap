// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyLoadImageContainer component should match the snapshot and styles 1`] = `
<div
  data-test-id="lazy-load-image"
>
  <t
    afterLoad={[Function]}
    alt="beer image"
    beforeLoad={[Function]}
    delayMethod="throttle"
    delayTime={300}
    effect="blur"
    height={100}
    placeholderSrc={null}
    src="image-url"
    threshold={100}
    useIntersectionObserver={true}
    visibleByDefault={false}
    wrapperClassName=""
  />
</div>
`;
